src/osmium/replication/server.py

```diff
@@ -285,15 +283,14 @@ def get_state_info(self, seq=None, retries=2):
         """
         for _ in range(retries + 1):
             try:
-                response = self.open_url(self.make_request(self.get_state_url(seq)))
+                response = self.open_url(self.get_state_url(seq))
             except Exception as err:
                 LOG.debug("Loading state info %s failed with: %s", seq, str(err))
                 return None
 
             ts = None
             seq = None
-            line = response.readline()
-            while line:
+            for line in response.iter_lines():
```

Comment:
We run into a compatibility issue here. The `open_url` function is part of the public interface and may be overwritten. See #65 and #67 for details. That means the code must still work when an urllib response object is returned. If iter_lines() does not work with the urllib response then I think that compatibility can be achieved by writing a generator function that looks what kind of object is returned (or rather if it has a iter_lines() function or not) and use the appropriate iteration method.

Reply:
Do you mean that I should also keep function self.make_request()?

I tried to have code compatible to both urlib and requests, but didn't think on checking the existence of iter_lines() - I will update the code.

Reply:
In addition, my change must break the cookie option in pyosmium-get-changes. I will check if it is possible to add a test for this option.

Reply:
The idea behind the extra `make_request` was that it would force the custom user agent, no matter what `open_url` implements. It might be good to keep that behaviour. It avoids code duplication in custom `open_url` functions.

Reply:
I've added a commit to add compatibility with both requests and urllib.request. I'm not a big fan of this change, even if I understand that we need to keep support for people changing open_url() function.